### ISSUE TITLE
rubocop:disable end line is incorrect

### ISSUE THREAD

**Issue by vfonic** [bbatsov/rubocop#3899]
I added `# rubocop:disable Method/LineLength` above a method. All the code below that method (including that method) had the rule disabled. I expected only that method to have the rule disabled. Is this expected behavior?

```ruby
class Klass
  # rubocop:disable Metrics/LineLength
  def method
    to_s
  end

  def method2
    'This line is too long. This line is too long. This line is too long. This line is too long.'
  end
end

```
--------

If you uncomment above `## rubocop:disable Method/LineLength`, the `method2` method below doesn't report the error anymore.

## Expected behavior

Rubocop only disables the rule in the method above which the rubocop:disable comment is placed.

## Actual behavior

All the code in the file, below the rubocop:disable, has the rule disabled.

## RuboCop version

v0.46.0

```
$ rubocop -V
0.46.0 (using Parser 2.3.3.1, running on ruby 2.3.3 x86_64-darwin16)
```

**Comment by jonas054** [comment]
> Is this expected behavior?

Yes. The cop is disabled until it's enabled again with `rubocop:enabled`, or until the end of the file.

http://rubocop.readthedocs.io/en/latest/configuration/#disabling-cops-within-source-code
